src/libsystemd-network/network-internal.c

```diff
@@ -772,16 +769,14 @@ int deserialize_in6_addrs(struct in6_addr **ret, const char *string) {
 }
 
 void serialize_dhcp_routes(FILE *f, const char *key, sd_dhcp_route **routes, size_t size) {
-        unsigned i;
-
         assert(f);
         assert(key);
         assert(routes);
         assert(size);
 
         fprintf(f, "%s=", key);
 
-        for (i = 0; i < size; i++) {
+        for (unsigned i = 0; i < size; i++) {
```

Comment:
given that `size` is of `size_t` (which I think is the right type) `i` should probably be too